src/main/java/it/pagopa/pn/deliverypush/service/mapper/RecipientMapper.java

```diff
@@ -0,0 +1,173 @@
+                                    .province(physicalAddress.getProvince())
+                                    .zip(physicalAddress.getZip())
+                                    .build()
+                    );
+        }
+
+        it.pagopa.pn.delivery.generated.openapi.clients.delivery.model.NotificationPaymentInfo payment = recipient.getPayment();
+
+        if(payment != null){
+            NotificationPaymentInfoInt.NotificationPaymentInfoIntBuilder paymentInfoBuilder = NotificationPaymentInfoInt.builder()
+                    .noticeCode( payment.getNoticeCode() )
+                    .creditorTaxId( payment.getCreditorTaxId() )
+                    .pagoPaForm(
+                            NotificationDocumentInt.builder()
+                                    .digests(
+                                            NotificationDocumentInt.Digests.builder()
+                                                    .sha256(payment.getPagoPaForm().getDigests().getSha256())
+                                                    .build()
+                                    )
+                                    .ref(
+                                            NotificationDocumentInt.Ref.builder()
+                                                    .key(payment.getPagoPaForm().getRef().getKey())
+                                                    .versionToken(payment.getPagoPaForm().getRef().getVersionToken())
+                                                    .build()
+                                    )
+                                    .build()
+                    );
+
+            if(payment.getF24flatRate() != null){
+                paymentInfoBuilder
+                        .f24flatRate(
+                                NotificationDocumentInt.builder()
+                                        .digests(
+                                                NotificationDocumentInt.Digests.builder()
+                                                        .sha256(payment.getF24flatRate().getDigests().getSha256())
+                                                        .build()
+                                        )
+                                        .ref(
+                                                NotificationDocumentInt.Ref.builder()
+                                                        .key(payment.getF24flatRate().getRef().getKey())
+                                                        .versionToken(payment.getF24flatRate().getRef().getVersionToken())
+                                                        .build()
+                                        )
+                                        .build()
+                        );
+            }
+
+
+
+            notificationRecIntBuilder.payment(paymentInfoBuilder.build());
+        }
+        
+        return notificationRecIntBuilder.build();
+    }
+
+    public static NotificationRecipient internalToExternal(NotificationRecipientInt recipient) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 54 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 98% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test. We recommend that you simplify this method or break it into multiple methods.